Banking/src/main/java/com/example/Banking/Controller/UserController.java

```diff
@@ -0,0 +1,76 @@
+package com.example.Banking.Controller;
+
+import java.util.List;
+import java.util.Optional;
+
+
+import com.example.Banking.Exception.ResourceNotFoundException;
+import com.example.Banking.Model.User;
+import com.example.Banking.Repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+//import javax.validation.Valid;
+
+
+@RestController
+@RequestMapping("/Users")
+public class UserController {
+    @Autowired
+    private UserRepository userRepository;
+
+    @GetMapping("/GetAllUsers")
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    @GetMapping("/GetOneUser/{id}")
+    public Optional<User> getOneUser(@PathVariable(value = "id") Long userId) {
+        return userRepository.findById(userId);
+    }
+
+    @PostMapping("/CreateUser")
+    public User createUser(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    @DeleteMapping("/DeleteUser/{id}")
+    public void deleteUser(@PathVariable(value = "id") Long userId){
+        userRepository.deleteById(userId);
+    }
+
+    @PutMapping("/transaction/{id}")
+        public String transaction(@PathVariable(value = "id") Long userId, @RequestBody User userAmount) throws ResourceNotFoundException {
```

Comment:
By userAmount you mean receiver?

Reply:
userAmount represents the Sender. I will change it into senderDetails.

Reply:
Why the transaction is based on userId? It should be on from and to accounts. A user can have multiple accounts. Also better create a separate pojo which represents a transaction having fields like fromAccount, toAccount and amount to represent a transaction